src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -939,43 +943,53 @@ protected void GenerateFindFirstChar()
                 }
             }
 
+            // Load necessary locals
+            // int runtextpos = this.runtextpos;
+            // int runtextend = this.runtextend;
+            Mvfldloc(s_runtextposField, _runtextposLocal);
+            Mvfldloc(s_runtextendField, _runtextendLocal);
+            if (_code.RightToLeft)
+            {
+                Mvfldloc(s_runtextbegField, _runtextbegLocal!);
+            }
+
             // Generate length check.  If the input isn't long enough to possibly match, fail quickly.
             int minRequiredLength = _code!.Tree.MinRequiredLength;
```

Comment:
(nit) you can remove the `!` on `_code!` now. (if you have other reasons for changing the code again)